packages/react-compose/src/next/simplifyShorthand.tsx

```diff
@@ -0,0 +1,27 @@
+import * as React from 'react';
+import { GenericDictionary } from './types';
+
+/**
+ * Ensures that the given slots are represented using object syntax. This ensures that
+ * the object can be merged along with other objects.
+ * @param props - The incoming props
+ * @param shorthandPropNames - An array of prop names to apply simplification to
+ */
+export const simplifyShorthand = (props: GenericDictionary, shorthandPropNames?: string[]) => {
+  let newProps = props;
+
+  if (shorthandPropNames && shorthandPropNames.length) {
+    newProps = {
+      ...props,
+    };
```

Comment:
Isn't this exactly the same as the `newProps = props` you're doing above?

Reply:
spreading will ensure we don't make modifications to the original props. if I could get rid of spreading here, I would, but i don't see a clean way to avoid the duplicate.